src/service/requests/useGetAllMonitorRequests/index.tsx

```diff
@@ -0,0 +1,53 @@
+import { AxiosError } from 'axios';
+import { useState } from 'react';
+import { TypeStatus } from '../../../utils/constants';
+import api from '../../api';
+import {
+  TListMonitorsRequestsErrorResponse,
+  TListMonitorsRequestParams,
+  TListMonitorsRequestResponse,
+} from './types';
+
+const useGetAllMonitorRequests = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string>();
+  const [data, setData] = useState<TListMonitorsRequestResponse>();
+
+  const listRequests = async (params?: TListMonitorsRequestParams) => {
+    setIsLoading(true);
+    setData(undefined);
+    setError(undefined);
+
+    try {
+      const response = await api.get('/monitor/all', { params });
+
+      const dt = response.data as TListMonitorsRequestResponse;
+
+      dt.data = dt.data.filter(
+        (monitor) => monitor.status.status === TypeStatus.PENDING
+      );
+
+      setData(dt);
```

Comment:
Aqui você podia ter passado logo o data para o estado, assim não precisaria criar o memo lá no hook da tela:
```suggestion
      setData(dt.data);
```

Reply:
Mas se eu passar o data para esse estado, eu perco as informações do meta da response não? E eu preciso delas para a paginação
![Screenshot from 2022-12-23 11-43-14](https://user-images.githubusercontent.com/50144860/209362167-83e96723-904c-498b-8f30-a5a1f91593bf.png)

Reply:
ah, boa. Não tinha me tocado disso. Deixa assim mesmo então 🚀